src/material/tabs/paginated-tab-header.ts

```diff
@@ -246,6 +246,30 @@ export abstract class MatPaginatedTabHeader
     });
   }
 
+  /** A method responsible for sending any change that could affect layout about
+   * items to subscribers.
+   */
+  _getItemChanges() {
+    const itemsTracked = new Set<MatPaginatedTabHeaderItem>();
+    const itemContainerItemChanged = new Subject<void>();
+
+    const observer = new ResizeObserver(() => {
+      itemContainerItemChanged.next();
+    });
+
+    return merge(this._items.changes, itemContainerItemChanged).pipe(
+      startWith(this._items),
+      map(() => {
+        for (const item of this._items.toArray()) {
+          if (!itemsTracked.has(item)) {
+            observer.observe(item.elementRef.nativeElement);
```

Comment:
Rather than observing each individual element, we could observe the container that they're in.

Reply:
I think that would be observing this which would trigger more changes than needed I think since the container changing sizes doesn't actually change the contents of their boxes.

What do you think?

Reply:
I think that in most cases changing the size of the container will change the position of the ink bar. It's also worth it for the reduced complexity.

Reply:
The container itself is full width in a lot of cases; I just checked. The tabs are then centered within it (at least in the demo app) so this won't actually work. I'm happy to come up with something else if you think this would be a problem but ResizeObserver already batches all the changes into a single call so it's probably not a problem for performance either. 

I do see that ResizeObserver will remove old calls to the `observe` when duplicated though, so I removed the set checking. Does that help?